### PR TITLE
Freebsd update

### PR DESCRIPTION
**Pull request by cgzones**

### CODE REVIEW

File: freebsd/FreeBSDProcessList.c [hunk 1]
```diff
@@ -8,24 +8,27 @@ in the source distribution for its full text.
...
-#include <time.h>
-#include <unistd.h>
-#include <sys/types.h>
+#include <sys/stat.h>
 #include <sys/sysctl.h>
+#include <sys/types.h>
 #include <sys/user.h>
+#include <time.h>
+#include <unistd.h>
+#include <unistd.h>
```
Comment (BenBE) [inline freebsd/FreeBSDProcessList.c:23]: Sorts before `sys/*`

File: freebsd/FreeBSDProcessList.c [hunk 2]
```diff
@@ -311,7 +319,71 @@ static inline void FreeBSDProcessList_scanMemoryInfo(ProcessList* pl) {
...
+      if (dirFd < 0)
+         goto err1;
+
+      const struct dirent* entry;
+      while ((entry = readdir(dirPtr))) {
+         if (!String_startsWith(entry->d_name, "tty"))
+            continue;
+
+         struct stat info;
+         if (fstatat(dirFd, entry->d_name, &info, 0) < 0)
```
Comment (BenBE) [inline freebsd/FreeBSDProcessList.c:341]: Check with `HAVE_FSTATAT` and fall back to `fstat` if necessary …

File: freebsd/FreeBSDProcessList.c [hunk 3]
```diff
@@ -311,7 +319,71 @@ static inline void FreeBSDProcessList_scanMemoryInfo(ProcessList* pl) {
...
+         return;
+
+      int dirFd = dirfd(dirPtr);
+      if (dirFd < 0)
+         goto err2;
+
+      const struct dirent* entry;
+      while ((entry = readdir(dirPtr))) {
+         struct stat info;
+         if (fstatat(dirFd, entry->d_name, &info, 0) < 0)
```
Comment (BenBE) [inline freebsd/FreeBSDProcessList.c:368]: Check `HAVE_FSTATAT` and fallback …

File: freebsd/Platform.c [hunk 4]
```diff
@@ -6,34 +6,36 @@ in the source distribution for its full text.
...
-#include "Meter.h"
+
+#include <devstat.h>
+#include <math.h>
+#include <net/if.h>
+#include <net/if_mib.h>
+#include <sys/resource.h>
+#include <sys/sysctl.h>
+#include <sys/time.h>
+#include <sys/types.h>
```
Comment (BenBE) [inline freebsd/Platform.c:17]: Subdirs after parent …

**Review by BenBE — CHANGES_REQUESTED** [review]

File: freebsd/FreeBSDProcessList.c (hunk 2 above)
Comment (cgzones) [reply freebsd/FreeBSDProcessList.c]: Added a compatibility wrapper

**Comment by cgzones** [comment]
Travis passed at https://travis-ci.com/github/cgzones/htop/builds/196165735

**Pull request merged**